mesonbuild/wrap/wrap.py

```diff
@@ -427,6 +427,24 @@ def find_program_provider(self, names: T.List[str]) -> T.Optional[str]:
                 return wrap_name
         return None
 
+    def get_cargo_wraps(self, cratename: str) -> T.List[str]:
+        found = []
+        for wrap in self.wraps.values():
+            if wrap.get('method') != 'cargo':
+                continue
+            if not wrap.name.endswith('-rs'):
+                continue
+            s = wrap.name.split('-')
+            if len(s) < 3:
+                continue
+            s.pop() # -rs
+            s.pop() # version
+            wrap_cratename = '-'.join(s)
```

Comment:
Can wraps using the `cargo` method have names not in the `$name-$version-rs` format? I don't think they can. If this is the case, the code can be simplified:
```python
for wrap in self.wrap.values():
     if wrap.get('method') == 'cargo':
         name, version, suffix = wrap.name.rsplit('-', 3)
         if name == crtatename:
             yield wrap.name

Reply:
Thanks for the recommendation, I'll use `rsplit('-', 2)` to extract crate_name.